sandbox-web/src/main/java/com/exadel/sandbox/team5/config/security/AuthenticationServiceImpl.java

```diff
@@ -20,9 +21,13 @@
 
     @Override
     public void authenticate(String login, String password) {
-        if (!authenticationManager.authenticate(
-                new UsernamePasswordAuthenticationToken(login, password)).isAuthenticated()) {
-            throw new BadCredentialsException("Incorrect username or password");
+        try {
+            if (!authenticationManager.authenticate(
+                    new UsernamePasswordAuthenticationToken(login, password)).isAuthenticated()) {
+                throw new BadCredentialsException("Incorrect username or password");
+            }
+        } catch (InternalAuthenticationServiceException e) {
```

Comment:
What is the root cause of the exception?

Reply:
public abstract class AuthenticationException extends RuntimeException

Reply:
use root exception?